presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -148,6 +149,7 @@ private static ConnectorPageSource createSelectivePageSource(
             DateTimeZone hiveStorageTimeZone,
             RowExpressionService rowExpressionService)
     {
+        columns = renumberPartitioningColumns(columns);
```

Comment:
in general, it is a bad practice to alter method parameters; consider changing the caller like so:

```
return createSelectivePageSource(selectivePageSourceFactories, configuration, session, hiveSplit, hiveLayout, assignUniqueIndicesToPartitionColumns(hiveColumns), hiveStorageTimeZone, rowExpressionService);
```

or

```
            List<HiveColumnHandle> hiveColumnsWithUniqueIndices = assignUniqueIndicesToPartitionColumns(hiveColumns);
            return createSelectivePageSource(selectivePageSourceFactories, configuration, session, hiveSplit, hiveLayout, hiveColumnsWithUniqueIndices, hiveStorageTimeZone, rowExpressionService);
```